### PR TITLE
Address Elastic Net Classifier failures

### LINKED ISSUE #2289: Perf test and deep dive into Elastic Net
PR [2269](https://github.com/alteryx/evalml/pull/2269) put up a fix for the Elastic Net classifier that partially solves the issues we had for fraud/lead-scoring demos. However, we want to look further into this Elastic Net classifer. We have 2 main questions:
- Are we incurring a double shrinkage using this classifier, as specified [here](https://en.wikipedia.org/wiki/Elastic_net_regularization#Specification)? This could explain the poor performance we have been seeing.
- Can we use SKLearn's [impl of Elastic Net](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.ElasticNet.html) rather than our current use of SGDClassifier? How does this change the performance of our classifier (run perf tests here)

Discussion chain in the comments on [this doc](https://alteryx.quip.com/xOV6A0uLg7aX/Fraud-and-Lead-Scoring-Jupyter-Notebook-changes)

TODO:
- Perf test new Elastic Net
- Figure out which to use moving forward, and look more into the performance to ensure we aren't subjecting our estimator to double shrinkage

### PR DESCRIPTION
**Pull request by bchen1116**
fix #2289 #2281 

Deep dive doc [here](https://ayx.atlassian.net/wiki/spaces/PS/pages/537526285/Elastic+Net+Classifier+deep+dive)
**IMPORTANT PERF TEST UPDATE [here](https://ayx.atlassian.net/wiki/spaces/PS/pages/537526285/Elastic+Net+Classifier+deep+dive#Changing-alpha-to-0.05%3A)**

Ultimately, we find that `alpha=0.05, l1_ratio=0.5` are the best default values for the Elastic Net Classifier. Additionally, we limit the hyperparameter search space of this in order to avoid `NaN` values in the output, an issue that will remain until scikit-learn fixes [this issue](https://github.com/scikit-learn/scikit-learn/issues/17978).

Perf tests and studies in the deep dive doc above.

### CODE REVIEW

File: evalml/tests/model_understanding_tests/prediction_explanations_tests/test_algorithms.py [hunk 1]
```diff
@@ -168,13 +168,7 @@ def test_shap(
         is_binary = False
     else:
         training_data, y = X_y_regression
-    # TODO: Figure out why we need to change the params in order for shap to pass. Filed as issue 2281
-    if "Elastic Net" in estimator.name:
```
Comment (ParthivNaresh) [inline evalml/tests/model_understanding_tests/prediction_explanations_tests/test_algorithms.py:172]: Sweet!

**Review by ParthivNaresh — APPROVED** [review]
Excellent work with the docs, really comprehensive analysis and it's good to see the comparison between our SGDClassifier, scikit-learn's Elastic Net Model, and Logistic Regression Classifier w/ Elastic Net Penalty.
I think there's nothing wrong with updating our Elastic Net Regressor to match the same values. Do we have issues with `NaN`s showing up in predictions with `alpha<0.01`?

**Comment by bchen1116** [comment]
@ParthivNaresh yep, when we set `alpha<=0.01`, we started seeing `NaN` values appear in the predictions at various `l1_ratio` values. I set it to `0.05` since I saw no `NaN` values here for a variety of `l1_ratio` values and data.

**Review by chukarsten — APPROVED** [review]
This looks good.  Still a little concerned that we're trying to hit a moving target, but this does let us take out a TODO, so that's solid.

**Comment by bchen1116** [comment]
Closing this PR in preference for [this one](https://github.com/alteryx/evalml/pull/2420)

**Pull request closed without merging**